File: app/src/main/java/com/gpaddy/baseandroid/theu/adapter/historyAdapter.java
package com.gpaddy.baseandroid.theu.adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

//import com.gpaddy.baseandroid.R;
//import com.gpaddy.baseandroid.theu.ttvnpt.chitietTin;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;

import com.gpaddy.baseandroid.R;
import com.gpaddy.baseandroid.theu.ttvnpt.*;
import com.gpaddy.baseandroid.theu.model.*;
//import com.gpaddy.baseandroid.theu.model.Article;
//import com.gpaddy.baseandroid.theu.model.HistoryModel;
//import com.gpaddy.baseandroid.theu.model.Utils;

public class historyAdapter extends RecyclerView.Adapter<historyAdapter.ItemHoler> {
    Context context;
    ArrayList<HistoryModel>historyModels;

    public historyAdapter(Context context, ArrayList<HistoryModel> historyModels) {
        this.context = context;
        this.historyModels = historyModels;
    }

    @NonNull
    @Override
    public historyAdapter.ItemHoler onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v= LayoutInflater.from(parent.getContext()).inflate(R.layout.dstimkiem_adapter,null);
        ItemHoler itemHoler=new ItemHoler(v);
        return itemHoler;
    }

    @Override
    public void onBindViewHolder(@NonNull historyAdapter.ItemHoler holder, int position) {
        Article model=historyModels.get(position).getArticleHistory();
        Picasso.get().load(model.getUrlToImage()).into(holder.img);
        holder.tieude.setText(model.getTitle());
        holder.tgian.setText(Utils.DateFormat(model.getPublishedAt()));
        holder.tenbao.setText("vnexpress");
    }

    @Override
    public int getItemCount() {
        return historyModels.size();
    }
    public class ItemHoler extends RecyclerView.ViewHolder{
        ImageView img;
        TextView tenbao, tgian, tieude, socmt;
        public ItemHoler(@NonNull View itemView) {
            super(itemView);
            tgian = itemView.findViewById(R.id.tvtgiandstk);
            tieude = itemView.findViewById(R.id.tvtieudedstk);
            tenbao = itemView.findViewById(R.id.tvtenbaodstk);
            img = itemView.findViewById(R.id.imgdstk);
            socmt = itemView.findViewById(R.id.socmt);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Toast.makeText(context, "hahahahehehe", Toast.LENGTH_SHORT).show();
                    Intent intent=new Intent(context, chitietTin.class);
                    intent.putExtra("thongtintin",historyModels.get(getAdapterPosition()).getArticleHistory());
                    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                    context.startActivity(intent);
                }
            });

        }
    }
}

File: app/src/main/java/com/gpaddy/baseandroid/ui/adapter/HistoryAdapter.kt
//package com.gpaddy.baseandroid.ui.adapter
//
//import android.view.LayoutInflater
//import android.view.ViewGroup
//import androidx.recyclerview.widget.DiffUtil
//import androidx.recyclerview.widget.ListAdapter
//import androidx.recyclerview.widget.RecyclerView
//import com.orhanobut.logger.Logger
//
//import com.gpaddy.baseandroid.data.model.db.HistoryModel
//import com.gpaddy.baseandroid.data.room.DatabaseDao
//import com.gpaddy.baseandroid.databinding.ItemHistoryBinding
//
//class HistoryAdapter  constructor(
//    private val databaseDao: DatabaseDao
//) : ListAdapter<HistoryModel, RecyclerView.ViewHolder>(BaseDiffCallBack()) {
//
//    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecommendViewHolder {
//        return RecommendViewHolder(
//            ItemHistoryBinding.inflate(
//                LayoutInflater.from(parent.context), parent, false
//            )
//        )
//    }
//
//    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
//        val data = getItem(position)
//        (holder as RecommendViewHolder).bind(data)
//    }
//
//    inner class RecommendViewHolder(val binding: ItemHistoryBinding) :
//        RecyclerView.ViewHolder(binding.root) {
//        init {
//            binding.btnXoa.setOnClickListener {
//                binding.data?.let { his ->
//                    Logger.e("error $his");
//
////                    Log.e("sang", "xoa: $his")
//                    databaseDao.deleteHis(his)
//                }
//            }
//        }
//
//        fun bind(data1: HistoryModel) {
//            binding.apply {
//                data = data1
//                executePendingBindings()
//            }
//        }
//
//    }
//
//    private class BaseDiffCallBack : DiffUtil.ItemCallback<HistoryModel>() {
//
//        override fun areItemsTheSame(oldItem: HistoryModel, newItem: HistoryModel): Boolean {
//            return oldItem.id_his == newItem.id_his
//        }
//
//        override fun areContentsTheSame(oldItem: HistoryModel, newItem: HistoryModel): Boolean {
//            return oldItem == newItem
//        }
//    }
//
//}